workers.tf

```diff
@@ -4,7 +4,7 @@ resource "aws_autoscaling_group" "workers" {
   max_size             = "${lookup(var.worker_groups[count.index], "asg_max_size",lookup(var.workers_group_defaults, "asg_max_size"))}"
   min_size             = "${lookup(var.worker_groups[count.index], "asg_min_size",lookup(var.workers_group_defaults, "asg_min_size"))}"
   launch_configuration = "${element(aws_launch_configuration.workers.*.id, count.index)}"
-  vpc_zone_identifier  = ["${var.subnets}"]
+  vpc_zone_identifier  = ["${coalescelist(var.worker_subnets,var.subnets)}"]
```

Comment:
IIRC, this is how the placement of autoscaling instances into subnets is determined. Wouldn't we want this to be just `var.worker_subnets` in that case?

Reply:
The way that this works is that `coalescelist` will return the first non-empty list, therefore if the user passes in a `worker_subnets` list when creating the module it will not be empty so vpc_zone_identifier will be set to that list, otherwise, it will be set to the value of the `var.subnets` list. I wrote it like this to ensure that current functionality would not be broken (so anyone currently using this module can upgrade versions without it affecting them).

Reply:
Related to my comment above, if `worker_subnets` is a property of each `worker_group` list item, the `subnets` input would only apply to the cluster which means this could be yanked. In that case, the variable could probably stand to be renamed `cluster_subnets` for clarity. I'm less worried about making breaking changes right now as this module has only been live for a week. Bumping a major rev is sometimes necessary early on.